foxpuppet/windows/browser/navbar.py

```diff
@@ -0,0 +1,39 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from selenium.webdriver.common.by import By
+
+from foxpuppet.region import Region
+
+
+class NavBar(Region):
+
+    """Representation of the NavBar
+    :param selenium: WebDriver Object
+    :param root: Root WebElement
+    :type selenium:
+        :py:class:`~selenium.webdriver.remote.webdriver.WebDriver` object
+    :type root:
+        :py:class:`~selenium.webdriver.remote.webdriver.WebDriver` object
+    """
+
+    _root_locator = (By.ID, 'nav-bar')
+    _tracking_protection_shield_locator = (By.ID, 'tracking-protection-icon')
+
+    def __init__(self, selenium, *args, **kwargs):
+        self.selenium = selenium
+
+    @property
+    def is_tracking_shield_displayed(self):
+        """Tracking Protection shield
+        :returns:
+            :py:class:`~selenium.webdriver.remote.webdriver.WebDriver` object
+        :return type: object
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            if self.selenium.find_element(
+                *self._tracking_protection_shield_locator). \
+                    get_attribute('state'):
+                return True
+            return False
```

Comment:
> So regular `self.find_element()` ?

No, that's PyPOM. We could add a `find_element` method to our `Region` though to introduce a similar behaviour. Otherwise you'd want `self.root.find_element`.

> Otherwise I wonder why we need this method at all? Why can't a test not use the tracking protection shield property to retrieve the element, and then retrieve the attribute itself? I don't see a benefit here in wrapping that.

The concept of page objects (which this is based on) means the consumer doesn't need to understand the API of Selenium. It also means that theoretically Selenium could be substituted for another tool.